vault/resource_ssh_secret_backend_role.go

```diff
@@ -363,10 +371,16 @@ func sshSecretBackendRoleRead(d *schema.ResourceData, meta interface{}) error {
 		"allowed_critical_options", "allowed_domains",
 		"cidr_list", "allowed_extensions", "default_extensions",
 		"default_critical_options", "allowed_users_template",
-		"allowed_users", "default_user", "key_id_format",
-		"max_ttl", "ttl", "algorithm_signer",
+		"allowed_users", "default_user_template", "default_user",
+		"key_id_format", "max_ttl", "ttl", "algorithm_signer",
+	}
+
+	if provider.IsAPISupported(meta, provider.VaultVersion112) {
```

Comment:
Nice! We should probably also do a similar check in the write method. At present we might cause an error if we are running against a version less than 1.12, since we are always writing the new `default_user_template` to Vault

Reply:
It doesn't error since we are using GetOk. But I am going to go ahead and do a version check on write as well.